store/store.go

```diff
@@ -1156,12 +1157,24 @@ var download = func(name, downloadURL string, user *auth.UserState, s *Store, w
 		Method: "GET",
 		URL:    storeURL,
 	}
-	resp, err := s.doRequest(client, reqOptions, user)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
 
+	var resp *http.Response
+out:
+	for _, n := range downloadBackoffs {
+		// we do *not* want to reuse the connection in this case
+		r, err := s.doRequest(&http.Client{}, reqOptions, user)
+		if err != nil {
+			return err
+		}
+		defer r.Body.Close()
+
+		resp = r
+
+		if r.StatusCode != 500 {
```

Comment:
no EOFs at this stage; those would happen during the Copy (at which stage you need to retry the whole thing from the top). Would require an outer loop, or retrying the download call itself.